utils/post.js

```diff
@@ -294,7 +294,8 @@ function modifyReactionsPost(post) {
     }
 
     if (!isUserDataIncluded) {
-      next.user.data = {};
+      next.user = next.user ?? {};
+      next.user.data = next.user.data ?? {};
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The changes made here are good for preventing null reference errors. However, it's important to note that this will only prevent null values. If `next.user` or `next.user.data` is `undefined`, the code will still fail. To handle both `null` and `undefined`, you could use a simple truthy check instead.
```diff
-      next.user = next.user ?? {};
-      next.user.data = next.user.data ?? {};
+      next.user = next.user || {};
+      next.user.data = next.user.data || {};
```
This way, if `next.user` or `next.user.data` is either `null` or `undefined`, it will be initialized to an empty object.


<!-- This is an auto-generated comment by OSS CodeRabbit -->